src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -68,6 +68,10 @@ def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[st
                 'http_addr': daemon_spec.ip if daemon_spec.ip else ''
             })
 
+        if 'dashboard' in self.mgr.get('mgr_map')['modules'] and spec.initial_admin_password:
```

Comment:
NIT: it would be good to have a function such as `self.mgr.module_is_enabled(module_name)` instead of accessing the map directly.